Shark Cage/createACLinNewDesktop/createACLinNewDesktop/util.cpp

```diff
@@ -133,75 +133,106 @@ namespace util {
 			tokenDeconstructed.TokenDefaultDacl,
 			tokenDeconstructed.TokenSource
 		);
+
+		//disable needed privileges
+		changeTokenCreationPrivilege(false);
+
+		if(!NT_SUCCESS(status)) {
+			if (NT_ERROR(status)) {
+				wprintf(L"  Cannot construct a token\n");
+				return false;
+			}
+		}
+
+		token = newToken;
+		return true;
 	}
 
 }
+
+//private functions
+
 ULONG getCurrentSessionID() {

```

Comment:
this function does not seem to be used anywhere, can it be deleted?

Reply:
The reason for its existence is explained in coments in `util.h` ... if for whatever reson the themplate token aquisition will be changed from `OpenProcessToken` to `WtsQueryUserToken` this function will be needed